linera-execution/src/system.rs

```diff
@@ -613,49 +583,17 @@ where
                 )))?;
             }
             CreateApplication {
+                application_id,
                 bytecode_id,
-                parameters,
                 instantiation_argument,
-                required_application_ids,
             } => {
-                let id = UserApplicationId {
-                    bytecode_id,
-                    creation: context.next_message_id(txn_tracker.next_message_index()),
-                };
-                for application in required_application_ids.iter().chain(iter::once(&id)) {
-                    self.check_and_record_bytecode_blobs(&application.bytecode_id, txn_tracker)
-                        .await?;
-                }
-                self.registry
-                    .register_new_application(
-                        id,
-                        parameters.clone(),
-                        required_application_ids.clone(),
-                    )
+                self.check_and_record_bytecode_blobs(&bytecode_id, txn_tracker)
                     .await?;
-                // Send a message to ourself to increment the message ID.
-                let message = RawOutgoingMessage {
-                    destination: Destination::Recipient(context.chain_id),
-                    authenticated: false,
-                    grant: Amount::ZERO,
-                    kind: MessageKind::Protected,
-                    message: SystemMessage::ApplicationCreated,
-                };
-                outcome.messages.push(message);
-                new_application = Some((id, instantiation_argument.clone()));
-            }
-            RequestApplication {
-                chain_id,
-                application_id,
-            } => {
-                let message = RawOutgoingMessage {
-                    destination: Destination::Recipient(chain_id),
-                    authenticated: false,
-                    grant: Amount::ZERO,
-                    kind: MessageKind::Simple,
-                    message: SystemMessage::RequestApplication(application_id),
-                };
-                outcome.messages.push(message);
+                txn_tracker.replay_oracle_response(OracleResponse::Blob(BlobId::new(
```

Comment:
Is this what the "effect coutner/index" refers to?

Reply:
I believe we also considered counting the number of applications created so far in the chain (and later the number of subchains etc).

Reply:
I think for now we're doing a per block effect index, which means it should match the transaction index, AFAIU